firedrake/assemble.py

```diff
@@ -1745,7 +1745,7 @@ def _as_global_kernel_arg_output(_, self):
     if rank == 0:
         return op2.GlobalKernelArg((1,))
     elif rank == 1 or rank == 2 and self._diagonal:
-        V, = Vs
+        V = Vs[0]
```

Comment:
We have `V,` here to assert a singleton. Why is that now violated? What are the contents of `V[1:]`?

Reply:
With slate we get the two copies of the same space in Vs. For some reason `diagonal=True` will cause `len(indices) == 1` only for `ufl.Form`, but not for `slate.Tensor`. On both cases `len(form.arguments()) == 2`.

https://github.com/firedrakeproject/firedrake/blob/32611a79ed56978c19ca004a1633addf31724aa4/firedrake/assemble.py#L2214

Reply:
Could you use `V = pytools.single_valued(Vs)` then?